lwgenerate/aws/aws.go

```diff
@@ -0,0 +1,383 @@
+	AwsRegion string
+}
+
+// Create a new AWS sub account
+//
+// A subaccount consists of the profile name (which needs to match the executing machines aws configuration) and a
+// region for any new resources to be created in
+func NewAwsSubAccount(profile string, region string) AwsSubAccount {
+	return AwsSubAccount{AwsProfile: profile, AwsRegion: region}
+}
+
+type GenerateAwsTfConfigurationArgs struct {
+	// Should we configure Cloudtrail integration in LW?
+	Cloudtrail bool
+
+	// Should we configure CSPM integration in LW?
+	Config bool
+
+	// Supply an AWS region for where to find the cloudtrail resources
+	// TODO @ipcrm future: support split regions for resources (s3 one place, sns another, etc)
+	AwsRegion string
+
+	// Supply an AWS Profile name for the main account, only asked if configuring multiple
+	AwsProfile string
+
+	// Existing S3 Bucket ARN (Required when using existing cloudtrail)
+	ExistingCloudtrailBucketArn string
+
+	// Optionally supply existing IAM role details
+	ExistingIamRole *ExistingIamRoleDetails
+
+	// Existing SNS Topic
+	ExistingSnsTopicArn string
+
+	// Consolidated Trail
+	ConsolidatedCloudtrail bool
+
+	// Should we force destroy the bucket if it has stuff in it? (only relevant on new Cloudtrail creation)
+	ForceDestroyS3Bucket bool
+
+	// For AWS Subaccounts in consolidated CT setups
+	// TODO @ipcrm future: what about many individual ct/config integrations together?
+	SubAccounts []AwsSubAccount
+
+	// Lacework Profile to use
+	LaceworkProfile string
+}
+
+// Ensure all combinations of inputs our valid for supported spec
+func (args *GenerateAwsTfConfigurationArgs) validate() error {
+	// Validate one of config or cloudtrail was enabled; otherwise error out
+	if !args.Cloudtrail && !args.Config {
+		return errors.New("cloudtrail or config integration must be enabled")
+	}
+
+	// Validate that at least region was set
+	if args.AwsRegion == "" {
+		return errors.New("AWS region must be set")
+	}
+
+	// Validate existing role IAM values, if set
+	if args.ExistingIamRole != nil {
+		if args.ExistingIamRole.Arn == "" ||
+			args.ExistingIamRole.Name == "" ||
+			args.ExistingIamRole.ExternalId == "" {
+			return errors.New("when using an existing IAM role, existing role ARN, name, and external ID all must be set")
+		}
+	}
+
+	return nil
+}
+
+type AwsTerraformModifier func(c *GenerateAwsTfConfigurationArgs)
+
+// NewAwsTerraform returns an instance of the GenerateAwsTfConfigurationArgs struct with the provided region and enabled
+// settings (config/cloudtrail).
+//
+// Note: Additional configuration details may be set using modifiers of the AwsTerraformModifier type
+//
+// Basic usage: Initialize a new AwsTerraformModifier struct, with a non-default AWS profile set. Then use generate to
+//              create a string output of the required HCL.
+//
+//   hcl, err := aws.NewAwsTerraform(
+//     "us-east-1",
+//     true,
+//     true,
+//     aws.WithAwsProfile("mycorp-profile")).Generate()
```

Comment:
nit: no need to split this lines that much
```suggestion
//   hcl, err := aws.NewAwsTerraform("us-east-1", true, true,
//     aws.WithAwsProfile("mycorp-profile")).Generate()
```

Reply:
👍